instrumentation/http/src/main/java/brave/http/HttpParserAdapter.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2013-2020 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package brave.http;
+
+import brave.Span;
+import brave.SpanCustomizer;
+import brave.internal.Nullable;
+import brave.propagation.CurrentTraceContext;
+import brave.propagation.CurrentTraceContext.Scope;
+import brave.propagation.TraceContext;
+
+/** Adapts {@link HttpParser} to the new request and response objects. */
+@Deprecated final class HttpParserAdapter implements HttpRequestParser, HttpResponseParser {
```

Comment:
This class is mostly impossible to grok ;) But we can leave it at that since it's deprecated, I wonder if there isn't a small readability improvement possible though by splitting client and server.

Reply:
agreed